dotCMS/src/integration-test/java/com/dotcms/publishing/PublisherAPITest.java

```diff
@@ -0,0 +1,273 @@
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.business.RoleAPI;
+import com.dotmarketing.business.UserAPI;
+import com.dotmarketing.cms.factories.PublicEncryptionFactory;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.portlets.contentlet.business.HostAPI;
+import com.dotmarketing.portlets.folders.business.FolderAPI;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.portlets.languagesmanager.business.LanguageAPI;
+import com.dotmarketing.util.ConfigUtils;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.liferay.portal.model.User;
+import java.io.File;
+import java.io.FileFilter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Created by Oscar Arrieta on 7/7/17.
+ */
+public class PublisherAPITest extends IntegrationTestBase {
+
+    private static HostAPI hostAPI;
+    private static FolderAPI folderAPI;
+    private static UserAPI userAPI;
+    private static RoleAPI roleAPI;
+    private static EnvironmentAPI environmentAPI;
+    private static PublishingEndPointAPI publisherEndPointAPI;
+    private static PublisherAPI publisherAPI;
+    private static LanguageAPI languageAPI;
+    private static PublishAuditAPI publishAuditAPI;
+    private static BundleAPI bundleAPI;
+
+    private static User systemUser;
+    private static User adminUser;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+
+        LicenseTestUtil.getLicense();
+
+        hostAPI = APILocator.getHostAPI();
+        folderAPI = APILocator.getFolderAPI();
+        userAPI = APILocator.getUserAPI();
+        roleAPI = APILocator.getRoleAPI();
+        environmentAPI = APILocator.getEnvironmentAPI();
+        publisherEndPointAPI = APILocator.getPublisherEndPointAPI();
+        publisherAPI = APILocator.getPublisherAPI();
+        languageAPI = APILocator.getLanguageAPI();
+        publishAuditAPI = PublishAuditAPI.getInstance();
+        bundleAPI = APILocator.getBundleAPI();
+
+        systemUser = userAPI.getSystemUser();
+        adminUser = APILocator.getUserAPI()
+                .loadByUserByEmail("admin@dotcms.com", systemUser, false);
+    }
+
+    /**
+     * This test will:
+     * ---Setup sender environment.
+     * ---Get a random folder from the default host.
+     * ---Push publish the folder. (It will fail cause receiver doesn't exist)
+     * ---Get the date of a file.
+     * ---Push publish retry. (Retry #1)
+     * ---Check that the file dates
+     *
+     * https://github.com/dotCMS/core/issues/12038
+     */
+    @Test
+    public void test_publish_fail_retry() {
```

Comment:
@jdotcms should something in the lines of : test_methodToTest_expectedResult_condition,  or similar